Sources/Turf/GeoJSON.swift

```diff
@@ -0,0 +1,358 @@
+        
+        if var unkeyedContainer = try? container.nestedUnkeyedContainer(forKey: .features) {
+            while (!unkeyedContainer.isAtEnd) {
+                let feature = try unkeyedContainer.decode(Feature.self)
+                
+                if let geometryType = GeometryType(rawValue: feature.simplifiedGeometry!.type) {
+                    featureTypes.append(geometryType.featureType)
+                }
+            }
+        }
+        
+        if var mappedContainer = try? container.nestedUnkeyedContainer(forKey: .features) {
+            while (!mappedContainer.isAtEnd) {
+                let type = featureTypes[mappedContainer.currentIndex]
+                let feature = try mappedContainer.decode(type.self)
+                features.append(feature)
+            }
+        }
+        
+        self.features = features
+        
+        try super.init(from: decoder)
+    }
+    
+    public override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(self.properties, forKey: .properties)
+        try container.encode(self.features, forKey: .features)
+        
+        try super.encode(to: encoder)
+    }
+}
+
+public enum GeometryType: String {
+    case Point
+    case LineString
+    case Polygon
+    case MultiPoint
+    case MultiLineString
+    case MultiPolygon
+    
+    static let allValues: [GeometryType] = [.Point, .LineString, .Polygon, .MultiPoint, .MultiLineString, .MultiPolygon]
+    
+    var featureType: Feature.Type {
+        switch self {
+        case .LineString:
+            return LineStringFeature.self
+        case .Polygon:
+            return PolygonFeature.self
+        case .Point:
+            return PointFeature.self
+        case .MultiPoint:
+            return MultiPointFeature.self
+        case .MultiLineString:
+            return MultiLineStringFeature.self
+        case .MultiPolygon:
+            return MultiPolygonFeature.self
+        }
+    }
+}
+
+public enum GeoJSONType: String {
+    case Feature
+    case FeatureCollection
+    static let allValues: [GeoJSONType] = [.Feature, .FeatureCollection]
+}
+
+public enum GeoJSONError: Error {
+    case unknownType
+    case noTypeFound
+}
+
+
+
+public class GeoJSON: Codable {
+    
+    var value: Codable?
+    
+    public required init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let feature = try container.decode(Feature.self)
+        
+        if feature.type == .Feature {
+            guard let type = GeometryType(rawValue: feature.simplifiedGeometry!.type) else {
+                throw GeoJSONError.unknownType
+            }
+            self.value = try container.decode(type.featureType)
+        } else if feature.type == .FeatureCollection {
+            self.value = try container.decode(FeatureCollection.self)
+        } else {
+            throw GeoJSONError.noTypeFound
```

Comment:
RFC&nbsp;7946 allows a “GeoJSON text” (a data blob) to consist of a bare Geometry, even a bare Position. Does this GeoJSON class support deserializing a Polygon or `CLLocationCoordinate2D` at the top level?